be/src/exec/pipeline/pipeline_driver_executor.cpp

```diff
@@ -336,7 +337,13 @@ void GlobalDriverExecutor::report_exec_state(QueryContext* query_ctx, FragmentCo
     this->_exec_state_reporter->submit(std::move(report_task));
 }
 
-void GlobalDriverExecutor::report_audit_statistics(QueryContext* query_ctx, FragmentContext* fragment_ctx) {
+void GlobalDriverExecutor::report_audit_statistics(QueryContext* query_ctx, FragmentContext* fragment_ctx, bool* done) {
+    bool submit_success = false;
+    DeferOp defer([&]() {
+        if (!submit_success) {
```

Comment:
```suggestion
        if (submit_success) {
```

Reply:
This method need to guarantee that the done will be finally set to `true` in any cases.
* If sumit_success = true, then the done will be set inside the lambda.
* The defer here is to handle the case if the task is submit failed.

Reply:
Add some comments here.